index_impl.go

```diff
@@ -485,6 +485,8 @@ func (i *indexImpl) preSearch(ctx context.Context, req *SearchRequest, reader in
 	}
 
 	var fts search.FieldTermSynonymMap
+	var count uint64
+	fieldCardinality := make(map[string]int)
```

Comment:
Lets change this to ..
```
var fieldCardinality map[string]int
```

.. and `make` only if `isBM25Enabled(..) == true` in line 499 below.